New-EntraOauth2PermissionGrant GA and Beta (#872)
microsoftgraph/entra-powershell: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md

---
title: New-EntraOauth2PermissionGrant
description: This article provides details on the New-EntraOauth2PermissionGrant command.

ms.service: active-directory
ms.topic: reference
ms.date: 06/28/2024
ms.author: eunicewaweru
ms.reviewer: stevemutungi
manager: CelesteDG

external help file: Microsoft.Graph.Entra-help.xml
Module Name: Microsoft.Graph.Entra
online version:
schema: 2.0.0
---

# New-EntraOauth2PermissionGrant

## Synopsis

> Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.

Reviewer: ```suggestion
Create a delegated permission grant using an oAuth2PermissionGrant object. This grant allows a client service principal to access a resource service principal on behalf of a signed-in user, with access restricted to the specified delegated permissions.
```

## Syntax

```powershell
New-EntraOauth2PermissionGrant 
 -ClientId <string> 
 -ConsentType <string> 
 -ResourceId <string> 
 [-PrincipalId <string>]
 [-Scope <string>]  
 [<CommonParameters>]
```

## Description

> The `New-EntraOauth2PermissionGrant` cmdlet Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.

Reviewer: ```suggestion
The `New-EntraOauth2PermissionGrant` cmdlet creates a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the specified delegated permissions.
```

## Examples

### Example 1: To grant authorization to impersonate all users

```powershell

> Connect-Entra -Scopes 'Directory.ReadWrite.All'

Reviewer: ```suggestion
Connect-Entra -Scopes 'DelegatedPermissionGrant.ReadWrite.All'
```

> New-EntraOauth2PermissionGrant -ClientId '00001111-aaaa-2222-bbbb-3333cccc4444' -ConsentType 'AllPrincipals' -ResourceId 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1' -Scope 'DelegatedPermissionGrant.ReadWrite.All'

Reviewer: ```suggestion
$params = @{
    ClientId = '00001111-aaaa-2222-bbbb-3333cccc4444'
    ConsentType = 'AllPrincipals'
    ResourceId = 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1'
    Scope = 'DelegatedPermissionGrant.ReadWrite.All'
}

New-EntraOauth2PermissionGrant @params
```

```

```Output
Id                                          ClientId                             ConsentType   ExpiryTime          PrincipalId ResourceId                           Scope
--                                          --------                             -----------   ----------          ----------- ----------                           -----
A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1 AllPrincipals 28/06/2025 07:44:25             aaaaaaaa-bbbb-cccc-1111-222222222222 Dele...

```

This command Grant authorization to impersonate all users.

### Example 2: To grant authorization to impersonate a specific user

```powershell

> Connect-Entra -Scopes 'Directory.ReadWrite.All'

Reviewer: ```suggestion
Connect-Entra -Scopes 'DelegatedPermissionGrant.ReadWrite.All'
```

> New-EntraOauth2PermissionGrant -ClientId '00001111-aaaa-2222-bbbb-3333cccc4444' -ConsentType 'Principal' -PrincipalId 'aaaaaaaa-bbbb-cccc-1111-222222222222' -ResourceId 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1' -Scope 'DelegatedPermissionGrant.ReadWrite.All' 

Reviewer: ```suggestion
$params = @{
    ClientId = '00001111-aaaa-2222-bbbb-3333cccc4444'
    ConsentType = 'Principal'
    PrincipalId = 'aaaaaaaa-bbbb-cccc-1111-222222222222'
    ResourceId = 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1'
    Scope = 'DelegatedPermissionGrant.ReadWrite.All'
}

New-EntraOauth2PermissionGrant @params
```

```

```Output
Id                                          ClientId                             ConsentType   ExpiryTime          PrincipalId ResourceId                           Scope
--                                          --------                             -----------   ----------          ----------- ----------                           -----
A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1 AllPrincipals 28/06/2025 07:44:25             aaaaaaaa-bbbb-cccc-1111-222222222222 Dele...
```

This command Grant authorization to impersonate a specific user.

## Parameters

### -ClientId

The object ID (not appId) of the client service principal for the application, which is authorized to act on behalf of a signed-in user when accessing an API. Required. Supports $filter (eq only).

```yaml
Type: System.String
Parameter Sets: CreateExpanded
Aliases:
Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -ConsentType

> Indicates if the client application is authorized to impersonate all users or only a specific user. "AllPrincipals" indicates authorization to impersonate all users, while "Principal" indicates authorization to impersonate a specific user. Consent on behalf of all users can be granted by an administrator. In some cases, non admin users are authorized to consent on behalf of themselves for certain delegated permissions. This is required and supports $filter (eq only).

Reviewer: ```suggestion
Indicates whether the client application is authorized to impersonate all users or only a specific user.
- `AllPrincipals`: Authorizes the application to impersonate all users.
- `Principal`: Authorizes the application to impersonate a specific user.
An administrator can grant consent on behalf of all users. In some cases, non-admin users are authorized to consent on behalf of themselves for certain delegated permissions. This parameter is required and supports the $filter query (eq only).
```
